### ISSUE TITLE
Support INFORMATION_SCHEMA.COLUMNS.INTERVAL_TYPE in H2

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#10389]
When generating code for a table like this in H2:

```sql
CREATE TABLE interval_types (
  id integer NOT NULL,
  ym interval year to month,
  ds interval day to second, 
  
  CONSTRAINT pk_interval_types PRIMARY KEY (id)
);
```

We're currently generating `SQLDataType.INTERVAL` or `YearToSecond` interval types, not `YearToMonth` and `DayToSecond`. This is because we're only considering the `INFORMATION_SCHEMA.COLUMNS.TYPE_NAME` column right now:

```sql
select column_name, interval_type, type_name, column_type
from INFORMATION_SCHEMA.columns
where table_schema = 'PUBLIC' 
```

Yielding

```
COLUMN_NAME|INTERVAL_TYPE|TYPE_NAME|COLUMN_TYPE           |
-----------|-------------|---------|----------------------|
ID         |             |INTEGER  |INTEGER NOT NULL      |
YM         |YEAR TO MONTH|INTERVAL |INTERVAL YEAR TO MONTH|
DS         |DAY TO SECOND|INTERVAL |INTERVAL DAY TO SECOND|
```

**Comment by katzyn** [comment]
BTW, the next version of H2 will return only names in `INTERVAL_TYPE` as required by SQL/Schemata, in 1.4.200 this field may incorrectly contain leading field precision and / or fractional seconds precision.

**Comment by lukaseder** [comment]
Thanks, @katzyn. We currently don't support precision on interval types yet, so this shouldn't affect jOOQ.

**Comment by lukaseder** [comment]
Fixed as well for jOOQ 3.14. I will have a more in-depth look at the various RDBMS's interpretations of intervals via https://github.com/jOOQ/jOOQ/issues/2230. Currently, we can store intervals, but not read them from H2, which sends them not as the literal string `1-2`, but e.g. `INTERVAL '1-2' YEAR TO MONTH`

**Comment by katzyn** [comment]
The SQL Standard requires shortest `<literal>` here with additional requirement for placement of sign, not the `<unquoted interval string>`, so conversion from `INTERVAL` to `VARCHAR` in H2 seems to be Standard-compliant (unlike historic conversions from datetime data types).
